src/tabs/Pixnflix/index.tsx

```diff
@@ -1,57 +1,252 @@
 import React from 'react';
+import { Button, ButtonGroup, Divider, NumericInput } from '@blueprintjs/core';
+import { IconNames } from '@blueprintjs/icons';
+import {
+  ErrorLogger,
+  DEFAULT_WIDTH,
+  DEFAULT_HEIGHT,
+  DEFAULT_FPS,
+} from '../../bundles/pix_n_flix/types';
 
 type Props = {
   children?: never;
   className?: string;
   debuggerContext: any;
 };
 
-type State = {};
+type SideContentVideoDisplayMode = 'video' | 'still';
+
+type State = {
+  width: number;
+  height: number;
+  FPS: number;
+  mode: SideContentVideoDisplayMode;
+};
+
+type PixNFlix = {
+  toReplString: () => string;
+  init: (
+    video: HTMLVideoElement | null,
+    canvas: HTMLCanvasElement | null,
+    errorLogger: ErrorLogger
+  ) => void;
+  deinit: () => void;
+  startVideo: () => void;
+  snapPicture: () => void;
+  updateFPS: (fps: number) => void;
+  updateDimensions: (width: number, height: number) => void;
+};
 
 class Repeat extends React.Component<Props, State> {
   private $video: HTMLVideoElement | null = null;
 
   private $canvas: HTMLCanvasElement | null = null;
 
+  private pixNFlix: PixNFlix;
+
   constructor(props: any) {

```

Comment:
`constructor(props: any)` can we try to resolve the any typing?

Reply:
I believe the typing here is redundant since `React.Component<Props, State>` in line 40 would already have enforced that the props here in the constructor is of the `Props` type declared in line 11.